game/src/edit/roads.rs

```diff
@@ -398,14 +399,40 @@ fn make_main_panel(
     let available_lane_types_row = Widget::row(available_lane_types_row);
 
     let mut current_lanes_ltr = Vec::new();
-    for (id, _, lt) in road.lanes_ltr() {
-        // TODO Add direction arrow sometimes
+    for (id, dir, lt) in road.lanes_ltr() {
+        let mut stack = GeomBatchStack::vertical(vec![
+            Image::untinted(lane_type_to_icon(lt).unwrap())
+                .build_batch(ctx)
+                .unwrap()
+                .0,
+        ]);
+        if can_reverse(lt) {
+            stack.push(
+                Image::untinted(if dir == Direction::Fwd {
+                    "system/assets/edit/forwards.svg"
+                } else {
+                    "system/assets/edit/backwards.svg"
+                })
+                .build_batch(ctx)
+                .unwrap()
+                .0,
+            );
+        }
+        let stack_batch = stack.batch();
+        let stack_bounds = stack_batch.get_bounds();
+
         current_lanes_ltr.push(
             ctx.style()
-                .btn_plain
-                .icon(lane_type_to_icon(lt).unwrap())
+                .btn_solid_primary
+                .btn()
+                .image_batch(stack_batch, stack_bounds)
                 .disabled(Some(id) == current_lane)
-                .image_color(ctx.style().btn_outline.fg, ControlState::Disabled)
+                // We have to explicitly specify the source color as white; otherwise the entire
```

Comment:
@michaelkirk, with the default `RewriteColor::ChangeAll`, this happens:
![Screenshot from 2021-04-27 17-57-48](https://user-images.githubusercontent.com/1664407/116330452-6f4f7f00-a782-11eb-93b3-44dd9b0a1fbb.png)
I think it's because of the invisible rectangular background we add to get the padding to work:
https://github.com/a-b-street/abstreet/blob/71e559b6d5d7ea469f374653ef142db314bd10de/widgetry/src/widgets/image.rs#L232

Would it be crazy to change the rules for `ChangeAll` to not do anything when `Color::CLEAR` is the source?


And more generally, does this seem like the right general approach for building this slightly new card style? I'll probably wind up making a new `ButtonStyle` instead of abusing `btn_solid_primary` like this

Reply:
> Would it be crazy to change the rules for ChangeAll to not do anything when Color::CLEAR is the source?

I think this makes sense and would solve the problem. I can't imagine when you'd want to "change all" and have a clear part become not clear.

But if that did happen, you could redundantly and explicitly color the clear parts.

> And more generally, does this seem like the right general approach for building this slightly new card style? I'll probably wind up making a new ButtonStyle instead of abusing btn_solid_primary like this

I think I'd approach it more or less like you currently are.

A reminder that you can extract out a common builder if you're re-using a style locally, rather than adding a new top-level style to `ctx.style()` if you don't expect it to be used widely, e.g. like here https://github.com/a-b-street/abstreet/blob/master/game/src/sandbox/minimap.rs#L129